espnet/nets/pytorch_backend/transformer/attention.py

```diff
@@ -53,7 +53,16 @@ def forward_qkv(self, query, key, value):
         n_batch = query.size(0)
         q = self.linear_q(query).view(n_batch, -1, self.h, self.d_k)
         k = self.linear_k(key).view(n_batch, -1, self.h, self.d_k)
-        v = self.linear_v(value).view(n_batch, -1, self.h, self.d_k)
+
+        if expand_value:
+            v_shape = value.shape
+            v = (
+                self.linear_v(value[:1, :, :])
```

Comment:
What does `expand_value` do? It seems to select only the first sample?
It would be better to add some comments to explain this argument and its purpose.

Reply:
Thank you, I will add comments. Actually this is very important to reduce memory usage, because the batch size of the PAR might become too large. We compute all beams for all masks in a batched manner, so for example, if we have 10 beam size and 10 masks, we have `batch_size=100`.
So we compute this memory-heavy calculation only one time. In my local PC, applying this trick makes inference faster (3.2sec -> 2.1sec)

Reply:
Thanks for the explanation. Have you added some comments about `expand_value`? Sorry if I missed it somewhere.

Reply:
@Masao-Someki, can you add some comments about `expand_value` in this file?

Reply:
I realized that we need to expand the `key` as well, so I renamed it to `expand_kv`. I added the following comment to clarify this change:
https://github.com/Masao-Someki/espnet/blob/7eeb3c0608b63dd46cb4eaad3990c2d6ba1a948b/espnet/nets/pytorch_backend/transformer/attention.py#L121-L126